src/interface/blas1_interface.hpp

```diff
@@ -811,14 +811,41 @@ typename sb_handle_t::event_t _rotmg(
   auto d1_view = make_vector_view(_d1, inc, vector_size);
   auto d2_view = make_vector_view(_d2, inc, vector_size);
   auto x1_view = make_vector_view(_x1, inc, vector_size);
-  auto y1_view = make_vector_view(_y1, inc, vector_size);
   auto param_view = make_vector_view(_param, inc, param_size);
 
-  auto operation =
-      Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view, y1_view, param_view);
-  auto ret = sb_handle.execute(operation, _dependencies);
+  if constexpr (std::is_arithmetic_v<container_3_t>) {
+    constexpr helper::AllocType mem_type = std::is_pointer_v<container_0_t>
+                                               ? helper::AllocType::usm
+                                               : helper::AllocType::buffer;
+    auto _y1_tmp = blas::helper::allocate<mem_type, container_3_t>(
+        1, sb_handle.get_queue());
 
-  return ret;
+    auto copy_y1 =
+        blas::helper::copy_to_device(sb_handle.get_queue(), &_y1, _y1_tmp, 1);
+
+    auto y1_view = make_vector_view(_y1_tmp, inc, vector_size);
+    auto operation = Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view,
+                                              y1_view, param_view);
+#if !defined(AMD_GPU) && !defined(INTEL_GPU) && !defined(NVIDIA_GPU)
```

Comment:
These are tuning targets right? Should this be a runtime check instead?

Reply:
I don't expect that using a `TUNING_TARGET` different from the hardware in use will work. So personally I would not convert to a runtime check. With this check the downside is that if `DEFAULT` target is used and the device is not CPU it will be less performant, but to me it is ok since `DEFAULT` target doesn't care about performance.